client/src/reducers/index.ts

```diff
@@ -1,4 +1,29 @@
-import { createStore } from "redux";
-import { reducer } from "./Reducer";
+import {
+  applyMiddleware,
+  combineReducers,
+  createStore,
+  Middleware,
+} from "redux";
+import { reducer, ReducerProps } from "./Reducer";
+import { composeWithDevTools } from "redux-devtools-extension";
 
-export const store = createStore(reducer);
+export interface RootState {
+  reducer: ReducerProps;
+}
+
+const rootReducer = combineReducers({
+  reducer,
+});
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type, @typescript-eslint/explicit-module-boundary-types
+const configureStore = () => {
+  const middlewares: Middleware[] = [];
+  const middleWareEnhancer = applyMiddleware(...middlewares);
+  const store = createStore(
+    rootReducer,
+    composeWithDevTools(middleWareEnhancer)
+  );
+  return store;
+};
+
+export const store = configureStore();
```

Comment:
Because store is already declared in the anonymous function above, we are getting a tslint-noshadow-variable linting error. We can fix this by not declaring store variable here. But let's just merge this in for now and fix later since this PR is getting long.